packages/dashboard/src/components/robots/robot-summary.tsx

```diff
@@ -0,0 +1,281 @@
+
+    default:
+      return base.palette.warning.main;
+  }
+};
+
+const LinearProgressWithLabel = (props: LinearProgressProps & { value: number }) => {
+  return (
+    <Box sx={{ display: 'flex', alignItems: 'center' }}>
+      <Box sx={{ width: '100%', mr: 1 }}>
+        <LinearProgress variant="determinate" {...props} />
+      </Box>
+      <Box sx={{ minWidth: 35 }}>
+        <Typography variant="body2" color="text.secondary">{`${Math.round(
+          props.value,
+        )}%`}</Typography>
+      </Box>
+    </Box>
+  );
+};
+
+interface RobotSummaryProps {
+  onClose: () => void;
+  robot: RobotTableData;
+}
+
+const showBatteryIcon = (robot: RobotState, robotBattery: number) => {
+  if (robot.status === Status2.Charging) {
+    return <BatteryChargingFullIcon />;
+  }
+
+  const batteryIcons: Record<string, JSX.Element> = {
+    '100': <BatteryFullIcon />,
+    '80': <Battery60Icon />,
+    '40': <Battery20Icon />,
+    '0': <Battery20Icon />,
+  };
+
+  const key = Object.keys(batteryIcons).find((level) => robotBattery <= parseInt(level));
+
+  return key ? batteryIcons[key] : null;
+};
+
+export const RobotSummary = React.memo(({ onClose, robot }: RobotSummaryProps) => {
+  const classes = useStyles();
+  const rmf = React.useContext(RmfAppContext);
+
+  const [isOpen, setIsOpen] = React.useState(true);
+  const [robotState, setRobotState] = React.useState<RobotState | null>(null);
+  const [taskState, setTaskState] = React.useState<TaskState | null>(null);
+  const [openTaskDetailsLogs, setOpenTaskDetailsLogs] = React.useState(false);
+  const [navigationStart, setNavigationStart] = React.useState('-');
+  const [navigationDestination, setNavigationDestination] = React.useState('-');
+
+  React.useEffect(() => {
+    if (!rmf) {
+      return;
+    }
+    const sub = rmf
+      .getFleetStateObs(robot.fleet)
+      .pipe(
+        mergeMap((fleetState) => {
+          const robotState = fleetState?.robots?.[robot.name];
+          const taskObs = robotState?.task_id ? rmf.getTaskStateObs(robotState.task_id) : of(null);
+          return robotState ? combineLatest([of(robotState), taskObs]) : EMPTY;
+        }),
+      )
+      .subscribe(([robotState, taskState]) => {
+        setRobotState(robotState);
+        setTaskState(taskState);
+      });
+    return () => sub.unsubscribe();
+  }, [rmf, robot.fleet, robot.name]);
+
+  const taskProgress = React.useMemo(() => {
+    if (
+      !taskState ||
+      !taskState.estimate_millis ||
+      !taskState.unix_millis_start_time ||
+      !taskState.unix_millis_finish_time
+    ) {
+      console.log(`Can't calculate task progress`);
+      return undefined;
+    }
+
+    return Math.min(
+      1.0 -
+        taskState.estimate_millis /
+          (taskState.unix_millis_finish_time - taskState.unix_millis_start_time),
+      1,
+    );
+  }, [taskState]);
+
+  React.useEffect(() => {
+    if (!taskState || !taskState.phases || !taskState.active) {
+      setNavigationStart('-');
```

Comment:
```suggestion
      setNavigationStart(null);
```
as mentioned above, something like this.